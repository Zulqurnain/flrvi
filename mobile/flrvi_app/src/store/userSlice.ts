import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';
import { RootState } from './store';

const API_URL = 'http://localhost:5000/api/v1';

export const createProfile = createAsyncThunk<
  any,
  { birthday: any; gender: string; photos: any[] },
  { rejectValue: { msg: string } }
>('user/createProfile', async (profileData, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.post(`${API_URL}/profile`, profileData, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const updateUserProfile = createAsyncThunk<
  any,
  { bio: string; photos: string[] },
  { rejectValue: { msg: string } }
>('user/updateProfile', async (profileData, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.put(`${API_URL}/profile`, profileData, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const registerUser = createAsyncThunk<
  { token: string },
  { name: string; email: string; password: string },
  { rejectValue: { msg: string } }
>('user/register', async (userData, { rejectWithValue }) => {
  try {
    const response = await axios.post(`${API_URL}/users/register`, userData);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const loginUser = createAsyncThunk<
  { token: string },
  { email: string; password: string },
  { rejectValue: { msg: string } }
>('user/login', async (credentials, { rejectWithValue }) => {
  try {
    const response = await axios.post(`${API_URL}/users/login`, credentials);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const fetchProfiles = createAsyncThunk<
  any[],
  void,
  { rejectValue: { msg: string } }
>('user/fetchProfiles', async (_, { rejectWithValue }) => {
  try {
    const response = await axios.get(`${API_URL}/profile/search`);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const fetchConversations = createAsyncThunk<
  any[],
  void,
  { rejectValue: { msg: string } }
>('user/fetchConversations', async (_, { rejectWithValue }) => {
  try {
    const response = await axios.get(`${API_URL}/chat/conversations`);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const fetchUserProfile = createAsyncThunk<
  any,
  string,
  { rejectValue: { msg: string } }
>('user/fetchUserProfile', async (userId, { rejectWithValue }) => {
  try {
    const response = await axios.get(`${API_URL}/profile/user/${userId}`);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const fetchCurrentUserProfile = createAsyncThunk<
  any,
  void,
  { rejectValue: { msg: string } }
>('user/fetchCurrentUserProfile', async (_, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.get(`${API_URL}/profile/me`, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const likeUserProfile = createAsyncThunk<
  any,
  string,
  { rejectValue: { msg: string } }
>('user/likeUserProfile', async (userId, { rejectWithValue }) => {
  try {
    const response = await axios.post(`${API_URL}/profile/like/${userId}`);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const subscribeUser = createAsyncThunk<
  any,
  { email: string; plan: string; source: string },
  { rejectValue: { msg: string } }
>('user/subscribe', async (subscriptionData, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.post(`${API_URL}/payment/subscribe`, subscriptionData, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const purchaseBoost = createAsyncThunk<
  any,
  { productId: string; amount: number },
  { rejectValue: { msg: string } }
>('user/purchaseBoost', async (purchaseData, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.post(`${API_URL}/payment/purchase-consumable`, purchaseData, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const fetchBlockedUsers = createAsyncThunk<
  any[],
  void,
  { rejectValue: { msg: string } }
>('user/fetchBlockedUsers', async (_, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.get(`${API_URL}/users/blocked`, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const unblockUser = createAsyncThunk<
  any,
  string,
  { rejectValue: { msg: string } }
>('user/unblockUser', async (userId, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.post(`${API_URL}/users/unblock`, { userId }, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const updateNotificationSetting = createAsyncThunk<
  any,
  { [key: string]: boolean },
  { rejectValue: { msg: string } }
>('user/updateNotificationSetting', async (settings, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.post(`${API_URL}/users/settings/notifications`, settings, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const setLanguage = createAsyncThunk<
  any,
  string,
  { rejectValue: { msg: string } }
>('user/setLanguage', async (languageCode, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.post(`${API_URL}/users/settings/language`, { language: languageCode }, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

export const fetchTransactionHistory = createAsyncThunk<
  any[],
  void,
  { rejectValue: { msg: string } }
>('user/fetchTransactionHistory', async (_, { getState, rejectWithValue }) => {
  try {
    const { token } = (getState() as RootState).user;
    const config = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const response = await axios.get(`${API_URL}/payment/history`, config);
    return response.data;
  } catch (err: any) {
    return rejectWithValue(err.response.data);
  }
});

const userSlice = createSlice({
  name: 'user',
  initialState: {
    token: null,
    profile: null,
    profiles: [],
    conversations: [],
    blockedUsers: [],
    transactions: [],
    status: 'idle',
    error: null as string | null,
  },
  reducers: {
    logout: (state) => {
      state.token = null;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(loginUser.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(loginUser.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.token = action.payload.token;
      })
      .addCase(loginUser.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Login failed';
      })
      .addCase(registerUser.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(registerUser.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.token = action.payload.token;
      })
      .addCase(registerUser.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Registration failed';
      })
      .addCase(createProfile.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(createProfile.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.profile = action.payload;
      })
      .addCase(createProfile.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Profile creation failed';
      })
      .addCase(updateUserProfile.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(updateUserProfile.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.profile = action.payload;
      })
      .addCase(updateUserProfile.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Profile update failed';
      })
      .addCase(fetchProfiles.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchProfiles.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.profiles = action.payload;
      })
      .addCase(fetchProfiles.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Fetching profiles failed';
      })
      .addCase(fetchConversations.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchConversations.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.conversations = action.payload;
      })
      .addCase(fetchConversations.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Fetching conversations failed';
      })
      .addCase(fetchUserProfile.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchUserProfile.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.profile = action.payload;
      })
      .addCase(fetchUserProfile.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Fetching user profile failed';
      })
      .addCase(likeUserProfile.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(likeUserProfile.fulfilled, (state, action) => {
        state.status = 'succeeded';
      })
      .addCase(likeUserProfile.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Liking user profile failed';
      })
      .addCase(fetchCurrentUserProfile.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchCurrentUserProfile.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.profile = action.payload;
      })
      .addCase(fetchCurrentUserProfile.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Fetching current user profile failed';
      })
      .addCase(subscribeUser.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(subscribeUser.fulfilled, (state, action) => {
        state.status = 'succeeded';
        // Update user profile to reflect premium status
        if (state.profile) {
          state.profile.isPremium = true;
        }
      })
      .addCase(subscribeUser.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Subscription failed';
      })
      .addCase(purchaseBoost.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(purchaseBoost.fulfilled, (state, action) => {
        state.status = 'succeeded';
        // Update user profile to reflect boost status if needed
        // This could involve setting a boost expiration timestamp
      })
      .addCase(purchaseBoost.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Boost purchase failed';
      })
      .addCase(fetchBlockedUsers.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchBlockedUsers.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.blockedUsers = action.payload;
      })
      .addCase(fetchBlockedUsers.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Failed to fetch blocked users';
      })
      .addCase(unblockUser.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(unblockUser.fulfilled, (state, action) => {
        state.status = 'succeeded';
        // Remove the unblocked user from the blocked users list
        if (state.blockedUsers) {
          state.blockedUsers = state.blockedUsers.filter((user: any) => user.id !== action.meta.arg);
        }
      })
      .addCase(unblockUser.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Failed to unblock user';
      })
      .addCase(updateNotificationSetting.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(updateNotificationSetting.fulfilled, (state, action) => {
        state.status = 'succeeded';
        // Update the user's notification settings
        if (state.profile) {
          state.profile.notificationSettings = {
            ...state.profile.notificationSettings,
            ...action.meta.arg
          };
        }
      })
      .addCase(updateNotificationSetting.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Failed to update notification settings';
      })
      .addCase(setLanguage.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(setLanguage.fulfilled, (state, action) => {
        state.status = 'succeeded';
        // Update the user's language preference
        if (state.profile) {
          state.profile.language = action.meta.arg;
        }
      })
      .addCase(setLanguage.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Failed to set language';
      })
      .addCase(fetchTransactionHistory.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchTransactionHistory.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.transactions = action.payload;
      })
      .addCase(fetchTransactionHistory.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.payload ? (action.payload as { msg: string }).msg : 'Failed to fetch transaction history';
      });
  },
});

export const { logout } = userSlice.actions;
export const selectUserRegistrationStatus = (state: RootState) => state.user.status;
export const selectSubscriptionStatus = (state: RootState) => state.user.profile?.isPremium || false;
export const selectBoostPurchaseStatus = (state: RootState) => state.user.status;
export const selectBlockedUsers = (state: RootState) => state.user.blockedUsers || [];
export const selectNotificationSettings = (state: RootState) => state.user.profile?.notificationSettings || {};
export const selectCurrentLanguage = (state: RootState) => state.user.profile?.language || 'en';
export const selectAvailableLanguages = () => [
  { code: 'en', name: 'English' },
  { code: 'th', name: 'Thai' }
];
export const selectTransactionHistory = (state: RootState) => state.user.transactions || [];
export default userSlice.reducer;
