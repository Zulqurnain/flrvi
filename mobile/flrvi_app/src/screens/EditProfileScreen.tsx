import React, { useState, useEffect } from 'react';
import { View, Text, StyleSheet, ScrollView, TextInput, Button, Image, TouchableOpacity } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { updateUserProfile } from '../store/userSlice';
import { AppDispatch, RootState } from '../store/store';
import Header from '../components/Header';
import { launchImageLibrary } from 'react-native-image-picker';

const EditProfileScreen = ({ navigation }) => {
  const dispatch = useDispatch<AppDispatch>();
  const { profile } = useSelector((state: RootState) => state.user);
  const [bio, setBio] = useState('');
  const [photos, setPhotos] = useState<string[]>([]);

  useEffect(() => {
    if (profile) {
      setBio(profile.bio);
      setPhotos(profile.photos);
    }
  }, [profile]);

  const handleSave = () => {
    const profileData = { ...profile, bio, photos };
    dispatch(updateUserProfile(profileData)).then(() => {
      navigation.goBack();
    });
  };

  const handlePhotoUpload = () => {
    launchImageLibrary({ mediaType: 'photo', selectionLimit: 6 }, (response) => {
      if (response.didCancel) {
        console.log('User cancelled image picker');
      } else if (response.errorCode) {
        console.log('ImagePicker Error: ', response.errorMessage);
      } else if (response.assets) {
        const uris = response.assets.map((asset) => asset.uri).filter((uri): uri is string => !!uri);
        setPhotos(uris);
      }
    });
  };

  return (
    <ScrollView style={styles.container}>
      <Header
        title="Edit Profile"
        leftAction={{ icon: 'back', action: () => navigation.goBack() }}
        rightAction={{ label: 'Save', action: handleSave }}
      />
      <View style={styles.section}>
        <Text style={styles.sectionTitle}>Your Photos</Text>
        <Button title="Upload Photos" onPress={handlePhotoUpload} />
        <View style={styles.photoGrid}>
          {photos.map((uri, index) => (
            <Image key={index} source={{ uri }} style={styles.thumbnail} />
          ))}
        </View>
      </View>
      <View style={styles.section}>
        <Text style={styles.sectionTitle}>About Me</Text>
        <TextInput
          style={styles.textArea}
          value={bio}
          onChangeText={setBio}
          maxLength={500}
          multiline
        />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5F5F5',
  },
  section: {
    padding: 20,
    backgroundColor: '#FFFFFF',
    marginTop: 10,
  },
  sectionTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 10,
    color: '#333333',
  },
  textArea: {
    height: 150,
    borderColor: '#BDBDBD',
    borderWidth: 1,
    borderRadius: 8,
    padding: 10,
    fontSize: 16,
    textAlignVertical: 'top',
  },
  photoGrid: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    marginTop: 20,
  },
  thumbnail: {
    width: 100,
    height: 100,
    margin: 5,
  },
});

export default EditProfileScreen;
