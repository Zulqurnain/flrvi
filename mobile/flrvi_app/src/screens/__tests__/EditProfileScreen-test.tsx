import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import EditProfileScreen from '../EditProfileScreen';
import { updateUserProfile } from '../../store/userSlice';

const mockStore = configureStore([]);
const mockNavigation = { goBack: jest.fn() };

describe('EditProfileScreen', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      user: {
        profile: {
          bio: 'I am a software engineer.',
          photos: [],
        },
        status: 'idle',
        error: null,
      },
    });

    component = (
      <Provider store={store}>
        <EditProfileScreen navigation={mockNavigation} />
      </Provider>
    );
  });

  it('renders correctly with initial profile data', () => {
    const { getByDisplayValue } = render(component);
    expect(getByDisplayValue('I am a software engineer.')).toBeTruthy();
  });

  it('dispatches an action to update the user profile when the form is submitted', () => {
    const { getByText, getByDisplayValue } = render(component);

    fireEvent.changeText(getByDisplayValue('I am a software engineer.'), 'I am a product manager.');

    fireEvent.press(getByText('Save'));

    const expectedAction = updateUserProfile({
      bio: 'I am a product manager.',
      photos: [],
    });

    const actions = store.getActions();
    expect(actions).toContainEqual(expectedAction);
  });
});
