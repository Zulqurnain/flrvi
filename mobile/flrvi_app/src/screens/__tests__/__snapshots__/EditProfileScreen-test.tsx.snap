// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "elevation": 5,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "padding": 16,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        }
      }
    >
      <View
        style={
          {
            "width": 50,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <
            color="black"
            name="back"
            size={24}
          />
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Edit Profile
        </Text>
      </View>
      <View
        style={
          {
            "width": 50,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text>
            Save
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "marginTop": 10,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#333333",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Your Photos
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Upload Photos
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginTop": 20,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "marginTop": 10,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#333333",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        About Me
      </Text>
      <TextInput
        maxLength={500}
        multiline={true}
        onChangeText={[Function]}
        style={
          {
            "borderColor": "#BDBDBD",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 16,
            "height": 150,
            "padding": 10,
            "textAlignVertical": "top",
          }
        }
        value=""
      />
    </View>
  </View>
</RCTScrollView>
`;
